sim-lib/src/lib.rs

```diff
@@ -674,12 +717,21 @@ impl Simulation {
 
             // Generate a consumer for the receiving end of the channel. It takes the event receiver that it'll pull
             // events from and the results sender to report the events it has triggered for further monitoring.
-            tasks.spawn(consume_events(
-                node.clone(),
-                receiver,
-                output_sender.clone(),
-                self.shutdown_trigger.clone(),
-            ));
+            // ce: consume event
+            let ce_listener = self.shutdown_listener.clone();
+            let ce_shutdown = self.shutdown_trigger.clone();
+            let ce_output_sender = output_sender.clone();
+            let ce_node = node.clone();
+            tasks.spawn(async move {
+                log::debug!("Starting events consumer for {}.", ce_node.lock().await.get_info());
+                match consume_events(ce_node, receiver, ce_output_sender, ce_listener).await {
+                    Ok(_) => log::debug!("Event consumer exited due to a triggered shutdown or because all channel senders have dropped."),
+                    Err(e) => {
+                        ce_shutdown.trigger();
+                        log::error!("Event consumer exited with error: {e:?}.");
+                    },
+                }
```

Comment:
Only learning best practises for these patterns myself recently, but I think you can use `if let Err(e) = consume ... else` here because we don't care about the actual `Ok` value.  

Also nitty: the `Ok` path log is a little long, I think we can just say `Event consumer exited.` since the only people looking at this level of logs is us, and we'll then grep for it in the codebase.